conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py

```diff
@@ -68,19 +71,36 @@ def run(known_args, pipeline_args):
   with beam.Pipeline(options=pipeline_options) as p:
     pcoll = (p | ReadFromText(known_args.input)
                    | beam.ParDo(WriteDecompressedFile())
-                   | beam.io.Write(bigquery.BigQuerySink(known_args.table, schema=known_args.schema)))
+                   | beam.io.Write(bigquery.BigQuerySink(known_args.table, schema=known_args.schema, validate = True)))
 
 def truncate_content(s):
     dic = json.loads(s) 
     dic['truncated'] = False
-    if sys.getsizeof(s) > THERESHOLD:
+    dic['no_records'] = 1
+    dic['record_index'] = 0
+    filesize = sys.getsizeof(s)
+    if filesize > THERESHOLD:
```

Comment:
Tip: if you have a simple condition, like the case when filesize <= THRESHOLD, where you return right away, put that case first. This makes it easier to read the code because you don't have to worry about there being an extra case as you read the complex case. Alternatively, you can take the complex case and make a helper function for it.

Reply:
Done, thanks for the tip! :)